vtex-apps/glovo-integration
# Create order

#### What problem is this solving?

This PR add a feature to create a VTEX order when receiving an order from the Glovo app.

#### How to test it?

To test it, create an order in the [Glovo test enviroment](https://testglovo.com/es/en/barcelona/ametller-origen-test) by adding either of these products:

- 100	aranja rosa
- 101	llimona
- 140	Carbassó extra
- 1428	Torrades artesanals Espiga Blanca 100g
- 1001	Halloumi Dodoni 225g

You will have to create an account with your email (might ask for phone number too). On checkout use the 'Pay with cash' method and use 'Carrer de Pallars, 500' as an address.


After placing the order, check it on [this workspace](https://glovo--ametllerorigenqa.myvtex.com/admin/checkout/)

You can check the order status update at [Splunk](https://splunk7.vtex.com/) with this query:
index=io_vtex_logs app="vtex.glovo-integration@0.0.1" account=ametllerorigenqa workspace=glovo level=error OR level=info


#### To keep in mind:

Since we are not yet handling edge cases, it might be good to note that I noticed that when a order being created fails, we get a messages array inside the returned simulation which contains what went wrong.


The orders are not created when the Last Name of the client is empty or undefined. Glovo only sends a name field which has to split, but they accept to have only one name in this field. That's why you'll see that I set  ['Name' and 'Lastname'](https://github.com/vtex-apps/glovo-integration/compare/create-order?expand=1#diff-2cbae0ae249e9cdc0379e8728c4fa0a717790493e81505d728b5f5c3333137beR96-R97) as a backup. Might want to find a better way to deal with this.

We'll have to set some of the variables as dynamic depending on whether the app is Linked in a dev workspace or is installed in master.

#### How does this PR make you feel?


![catjam](https://user-images.githubusercontent.com/17585823/115056790-891bd700-9ee3-11eb-8f2d-24cd2ee5a6d7.gif)
